src/java/fr/paris/lutece/plugins/workflow/modules/formspdf/service/HtmlToPDFGenerator.java

```diff
@@ -195,23 +283,185 @@ private void writeExportFile( Path directoryFile ) throws IOException
         }
 
     }
-    
     private void markersToModel( Map<String, Object> model, Collection<InfoMarker> collectionInfoMarkers )
     {
         for ( InfoMarker infoMarker : collectionInfoMarkers )
         {
-            model.put( infoMarker.getMarker(), infoMarker.getValue() );           
+            model.put( infoMarker.getMarker(), infoMarker.getValue() );
         }
     }
 
+    private Map<String, Object> markersToModel( Map<String, Object> model, Collection<InfoMarker> collectionInfoMarkers, List<FormQuestionResponse> formQuestionResponseList )
+    {
+        HashMap<Integer, FormQuestionResponse> formResponseListByEntryId = formResponseListToHashmap(formQuestionResponseList);
+        String adminBaseUrl = "";
+        if(StringUtils.isBlank(AppPropertiesService.getProperty( "lutece.admin.prod.url" )))
+        {
+            adminBaseUrl = AppPropertiesService.getProperty( "lutece.admin.prod.url" );
+        }
+        else
+        {
+            AppLogService.info( "lutece.admin.prod.url property not found" );
+            adminBaseUrl = _baseUrl;
+        }
+        for ( InfoMarker infoMarker : collectionInfoMarkers )
+        {
+            model.put( infoMarker.getMarker(), infoMarker.getValue() );
+            if(infoMarker.getMarker().contains("position_"))
+            {
+                String position = infoMarker.getMarker().replace("position_", "");
+                int positionInt = Integer.parseInt(position);
+                List<String> responseValue = getResponseValue(formResponseListByEntryId.get(positionInt));
+
+                String responseValueString = "";
+                for (String response : responseValue)
+                {
+                    responseValueString += response + " ";
+                }
+                model.put( infoMarker.getMarker(), responseValueString );
+
+            }
+            if(infoMarker.getMarker().equals("url_admin_forms_response_detail"))
+            {
+                String linkMessage = I18nService.getLocalizedString( LINK_MESSAGE_BO, Locale.getDefault( ) );
+                model.put( infoMarker.getMarker(), "<a href=\""+ adminBaseUrl+"/jsp/admin/plugins/forms/ManageDirectoryFormResponseDetails.jsp?view=view_form_response_details&id_form_response=" + _formResponse.getId( ) + "\">" + linkMessage + "</a>" );
+            }
+            if(infoMarker.getMarker().equals("url_fo_forms_response_detail"))
+            {
+                String linkMessage = I18nService.getLocalizedString( LINK_MESSAGE_FO, Locale.getDefault( ) );
+                model.put( infoMarker.getMarker(), "<a href=\""+ _baseUrl+"/jsp/site/Portal.jsp?page=formsResponse&view=formResponseView&id_response=" + _formResponse.getId( ) + "\">" + linkMessage + "</a>" );
+            }
+            if(infoMarker.getMarker().equals("creation_date"))
+            {
+                String creationDate = _formResponse.getCreation( ).toLocalDateTime().toString();
```

Comment:
Its better to send the date object, and not force the locale,  to let the template define the format

Reply:
There is no html template. 
The user is creating it with the tinymice editor and copy/past the markers.